### PR TITLE
Add docker image to installation

### LINKED ISSUE #751: Add kpt Docker image to the installation doc
https://googlecontainertools.github.io/kpt/installation/

Using the kpt docker image is another way to use kpt.

### PR DESCRIPTION
**Pull request by prachirp**
- Add docker image to installation
- Prune unpolished catalog examples
- Remove outdated helm usage doc
- Lint automatically when regenerating docs
- Fix some markdownlint warnings

Fixes #751

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,3 +55,4 @@ See the [docs] for more information on how to use `kpt`.
 [windows]: https://storage.googleapis.com/kpt-dev/latest/windows_amd64/kpt.exe
 [docs]: https://googlecontainertools.github.io/kpt
 [release]: https://github.com/GoogleContainerTools/kpt/releases/latest
+[gcr.io/kpt-dev/kpt]: https://pantheon.corp.google.com/gcr/images/kpt-dev/GLOBAL/kpt/
```
Comment (mortent) [inline README.md:58]: Can we use a link there that doesn't refer to internal google names?

File: site/content/en/guides/consumer/function/export/circleci/_index.md [hunk 2]
```diff
@@ -105,11 +105,11 @@ workflows:
 
 If you don’t have one yet, you can just copy the exported `config.yml` file into `.circleci/config.yml` in your project root. Then do the steps above to make the pipeline fully functional. Once all changes are pushed into GitHub, you can do the following steps to setting up your project on CircleCI:
 
-1.  Log into [CircleCI](https://circleci.com/) and choose `Log In with GitHub`.
-1.  Select an organization if prompted.
-1.  Choose your newly created repo and click `Set Up Project`.
-1.  Click `Add Manually` since you have already added `.circleci/config.yml`.
-1.  Click `Start Building`.
+1. Log into [CircleCI](https://circleci.com/) and choose `Log In with GitHub`.
```
Comment (mortent) [inline site/content/en/guides/consumer/function/export/circleci/_index.md]: Is this supposed to be a numbered list?

File: site/content/en/guides/consumer/function/export/circleci/_index.md [hunk 3]
```diff
@@ -72,9 +72,9 @@ version: "2.1"
 
 To merge the exported file with your existing pipeline, you can:
 
-1.  Copy and paste the `orbs` field
-1.  Insert a `checkout` step as the first step in the `run-functions` job.
-1.  Add `kpt/run-functions` to your workflow jobs.
+1. Copy and paste the `orbs` field
```
Comment (mortent) [inline site/content/en/guides/consumer/function/export/circleci/_index.md]: Should all items have the number `1`?

File: README.md (hunk 1 above)
Comment (prachirp) [reply README.md:58]: Nice catch, updated link in both README and installation page on the site.

File: site/content/en/guides/consumer/function/export/circleci/_index.md (hunk 3 above)
Comment (prachirp) [reply site/content/en/guides/consumer/function/export/circleci/_index.md]: Yes. "To be practical and avoid errors on the numbers, use "1" for all the items. The markdown engine will output them in the correct order." from https://about.gitlab.com/handbook/markdown-guide/#ordered-lists

File: site/content/en/guides/consumer/function/export/circleci/_index.md (hunk 2 above)
Comment (prachirp) [reply site/content/en/guides/consumer/function/export/circleci/_index.md]: Yes, the numbers get rendered correctly later.

**Pull request merged**